projects/mupdf/pdf_fuzzer.cc

```diff
@@ -17,11 +17,101 @@
 */
 
 #include <cstdint>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
 
 #include <mupdf/fitz.h>
 
+#define ALIGNMENT 16
+#define MAX_ALLOCATION (1024 * 1024 * 1024)
+
+static uint64_t total = 0;
+
+static void *
+fz_malloc_ossfuzz(void *opaque, size_t size)
+{
+	char *ptr = NULL;
+
+	if (size == 0)
+		return NULL;
+	if (size > SIZE_MAX - ALIGNMENT)
+		return NULL;
+
+	if (total + size + ALIGNMENT > MAX_ALLOCATION)
```

Comment:
this addition can potentially overflow and cause the check to incorrectly fail. should this be properly handled here?